eclipse-theia/theia
# Verify checksums while downloading plugins

#### What it does


Fixes: #8799 

The syntax of the URLs in `theiaPlugins` of `package.json` is extended so that we can define the expected integrity checksums in URL fragment - this way it remain compatible with existing usages (when there is no hash in URLs the checksums are not verified). The checksums are defined using https://developer.mozilla.org/en-US/docs/Web/Security/Subresource_Integrity hash format. The verification part is done using https://www.npmjs.com/package/ssri

This is a bit similar to what yarn uses for its `yarn.lock` file - but the usage is different, unlike yarn which automatically computes the signature on the first download, here we would have to compute and add "manually" the signatures in `package.json`.

#### How to test


The `package.json` have been modified in this PR to include the actual hashes. One way of testing can be to modify a signature in `package.json`, run `yarn download:plugins` and check that an error is reported because the downloaded plugin does not match signature.

#### Review checklist


#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
